### PR TITLE
Add a basic e2e test for 3rd party objects.

### PR DESCRIPTION
**Pull request by brendandburns**
Depends on https://github.com/kubernetes/kubernetes/pull/25365

[Analytics]()

### CODE REVIEW

File: test/e2e/third-party.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		ObjectMeta: api.ObjectMeta{
+			Name: "foo.company.com",
+		},
+		Versions: []extensions.APIVersion{
+			{Name: "v1"},
+		},
+	}
+
+	Context("Simple Third Party", func() {
+		It("creating/deleting thirdparty objects works [Conformance]", func() {
```
Comment (lavalamp) [inline test/e2e/third-party.go]: Is the conformance tag correct?

File: test/e2e/third-party.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			if out.Name != foo.Name || out.SomeField != foo.SomeField || out.OtherField != foo.OtherField {
+				framework.Failf("expected:\n%#v\nsaw:\n%#v\n%s\n", foo, &out, string(data))
+			}
+
+			data, err = f.Client.RESTClient.Get().AbsPath("/apis/company.com/v1/foos").DoRaw()
+			if err != nil {
+				framework.Failf("failed to list with no objects: %v", err)
+			}
+			if err := json.Unmarshal(data, &list); err != nil {
+				framework.Failf("failed to decode: %v", err)
```
Comment (lavalamp) [inline test/e2e/third-party.go]: %#v would be more helpful if this fails

**Comment by lavalamp** [comment]
LGTM, nits optional

File: test/e2e/third-party.go (hunk 1 above)
Comment (brendandburns) [inline test/e2e/third-party.go]: I believe so.  We want thirdparty to be a part of all working clusters, seems worth making it part of the conformance suite.

File: test/e2e/third-party.go (hunk 2 above)
Comment (brendandburns) [inline test/e2e/third-party.go]: done.

**Comment by brendandburns** [comment]
Comments addressed.

**Comment by brendandburns** [comment]
re-lgtm'd on @lavalamp said nits optional.

**Comment by brendandburns** [comment]
@k8s-bot unit test this please issue: #25965

**Pull request merged**